beacon_node/beacon_chain/src/light_client_optimistic_update_verification.rs

```diff
@@ -107,6 +111,16 @@ impl<T: BeaconChainTypes> VerifiedLightClientOptimisticUpdate<T> {
             None => return Err(Error::SigSlotStartIsNone),
         }
 
+        // check if we can process the optimistic update immediately
+        // otherwise queue
+        let canonical_root = light_client_optimistic_update
+            .attested_header
+            .canonical_root();
+
+        if canonical_root != head_block.message().parent_root() {
```

Comment:
Could we also get an update message where `update.attested_header` references a block which we have already seen? In this case, would it be better to simply check if we have seen `canonical_root` in fork choice instead of checking if it's equal to current head's parent? Might be completely wrong on this one, so please feel free to correct me :)

Reply:
I'm not certain. I don't think that would happen because the only time new optimistic updates are supposed to be published to pubsub is: "Whenever fork choice selects a new head block with a sync aggregate participation >= MIN_SYNC_COMMITTEE_PARTICIPANTS and a post-Altair parent block, full nodes with at least one validator assigned to the current sync committee at the block's slot". "If attested_header.slot increased, a LightClientOptimisticUpdate SHOULD be broadcasted to the pubsub topic light_client_optimistic_update if no matching message has not yet been forwarded as part of gossip validation." 
 https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/p2p-interface.md#sync-committee . That's where my understanding comes from but I might be thinking about it incorrectly.